chain/xrp/client/client.go

```diff
@@ -0,0 +1,275 @@
+	// No way to pass the amount in the input using legacy interface, so we estimate using min amount.
+	args, _ := xcbuilder.NewTransferArgs(from, to, xc.NewAmountBlockchainFromUint64(1))
+	return client.FetchTransferInput(ctx, args)
+}
+
+// SubmitTx submits a Template tx
+func (client *Client) SubmitTx(ctx context.Context, txInput xc.Tx) error {
+	return errors.New("not implemented")
+}
+
+// Returns transaction info - legacy/old endpoint
+func (client *Client) FetchLegacyTxInfo(ctx context.Context, txHash xc.TxHash) (xc.LegacyTxInfo, error) {
+	return xc.LegacyTxInfo{}, errors.New("not implemented")
+}
+
+// Returns transaction info - new endpoint
+func (client *Client) FetchTxInfo(ctx context.Context, txHash xc.TxHash) (xclient.TxInfo, error) {
+	return xclient.TxInfo{}, errors.New("not implemented")
+}
+
+// FetchBalance fetches token balance for a XRP address
+func (client *Client) FetchBalance(ctx context.Context, address xc.Address) (xc.AmountBlockchain, error) {
+	return client.FetchBalanceForAsset(ctx, address, client.Asset)
+}
+
+func (client *Client) FetchBalanceForAsset(ctx context.Context, address xc.Address, assetCfg xc.ITask) (xc.AmountBlockchain, error) {
+	switch asset := assetCfg.(type) {
+	case *xc.ChainConfig:
+		return client.FetchNativeBalance(ctx, address)
+	case *xc.TokenAssetConfig:
+		return client.fetchContractBalance(ctx, address, asset.Contract)
+	default:
+		contract := asset.GetContract()
+		logrus.WithFields(logrus.Fields{
+			"chain":      asset.GetChain().Chain,
+			"contract":   contract,
+			"asset_type": fmt.Sprintf("%T", asset),
+		}).Warn("fetching balance for unknown asset type")
+		return client.fetchContractBalance(ctx, address, contract)
+	}
+}
+
+// FetchNativeBalance fetches account native balance for a XRP address
+func (client *Client) FetchNativeBalance(ctx context.Context, address xc.Address) (xc.AmountBlockchain, error) {
+	zero := xc.NewAmountBlockchainFromUint64(0)
+
+	requestPayload := AccountInfoRequest{
+		Method: "account_info",
+		Params: []ParamEntry{
+			{
+				Account: address,
+			},
+		},
+	}
+
+	jsonPayload, err := json.Marshal(requestPayload)
+	if err != nil {
+		return zero, fmt.Errorf("failed to marshal request payload: %w", err)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "POST", client.Url, bytes.NewBuffer(jsonPayload))
+	if err != nil {
+		return zero, fmt.Errorf("failed to create new HTTP request: %w", err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	httpClient := http.DefaultClient
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return zero, fmt.Errorf("failed to execute HTTP request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return zero, fmt.Errorf("failed to fetch balance, HTTP status: %s", resp.Status)
+	}
+
+	var response AccountInfoResponse
+	if err := json.NewDecoder(resp.Body).Decode(&response); err != nil {
+		return zero, fmt.Errorf("failed to decode response body: %w", err)
+	}
```

Comment:
This code will be duplicated in all functions.  I recommend making adding a helper function that implements http functionality for XRP endpoints.  E.g. something like:

```go
func send(httpClient *http.Client, method string, inputData interface{}, outputData interface{}) (error) {}
```

Check out existing clients elsewhere.

https://github.com/cordialsys/crosschain/blob/main/chain/evm/client/client_beacon.go#L107

https://github.com/cordialsys/crosschain/blob/main/client/services/kiln/send.go